### PR TITLE
fix Conflict url for Time Series API

### PR DESCRIPTION
**Pull request by devProdigy**
## Description

Sometimes we might see error about conflicting urls from TS API.
This fix might not be the final in this case, we have to watch this errors ever happen aging.
Also, TODOs for refactoring were added to the code, cause if to do it in this PR it'll be mess.

Additionally:
- disable errors suppressing;
- tests were added;
- bump version.

[Azure DevOps Link](https://dev.azure.com/EquinorASA/OMNIA%20Prevent/_workitems/edit/17813)

## Type of change

Mark the relevant options (`[x]` will mark checkbox as checked).
- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [X] Tests
- [ ] Other

# Checklist:

- [ ] I have made corresponding changes to the documentation / README (if needed)
- [X] I have added tests that prove my fix is effective or that my feature works (if needed)

### CODE REVIEW

File: app/latigo/time_series_api/prediction_storage_provider.py [hunk 1]
```diff
@@ -61,11 +63,21 @@ def put_prediction(self, prediction_data: PredictionDataSet):
...
+                    asset_id=common_asset_id,
+                )
+            except HTTPError as error:
+                if error.response.status_code == 409:
+                    # if such tag_name might already exists in the TS try to get/create once more.
+                    meta, err = self.replace_cached_metadata_with_new(
+                        tag_name=output_tag_name, asset_id=common_asset_id, description=description
+                    )
+                else:
+                    raise
```
Comment (o-fedorov) [inline app/latigo/time_series_api/prediction_storage_provider.py:79]:
```suggestion
                if error.response.status_code != 409:
                    raise
                    
                # if such tag_name might already exists in the TS try to get/create once more.
                meta, err = self.replace_cached_metadata_with_new(
                    tag_name=output_tag_name, asset_id=common_asset_id, description=description
                )
```
Having a short-circuit break improves readability.

File: app/latigo/time_series_api/prediction_storage_provider.py [hunk 2]
```diff
@@ -61,11 +63,21 @@ def put_prediction(self, prediction_data: PredictionDataSet):
...
+
             if (not meta and not err) or err:
```
Comment (o-fedorov) [inline app/latigo/time_series_api/prediction_storage_provider.py:81]:
```suggestion
            if err or not meta:
```
I know it's not related to the PR, but cold not help myself

File: app/latigo/time_series_api/sensor_data_provider.py [hunk 3]
```diff
@@ -134,11 +134,7 @@ def get_data_for_range(
             ts, err = self._fetch_data_for_id(id, time_range)
             # logger.info(f" D '{ts}, {err}'")
             if err or not ts:
-                if ts:
-                    msg = ts.get("latigo-ok", "Unknown failure")
-                else:
-                    msg = "No ts"
-                return None, err or msg
+                return None, err or "No ts"
```
Comment (o-fedorov) [inline app/latigo/time_series_api/sensor_data_provider.py:137]:
```suggestion
                return None, (err or "No ts")
```
Just for better readability.

File: app/latigo/time_series_api/prediction_storage_provider.py (hunk 2 above)
Comment (devProdigy) [reply app/latigo/time_series_api/prediction_storage_provider.py:81]: I'll change it :)

File: app/latigo/time_series_api/prediction_storage_provider.py (hunk 1 above)
Comment (devProdigy) [reply app/latigo/time_series_api/prediction_storage_provider.py:79]: good point, thanks

**Pull request merged**